### PR TITLE
Validate Android package name and iOS bundle identifier

### LINKED ISSUE #7324: Validate package name on Android and Bundle Identifier on iOS/MacOS contains only valid symbols
Describe the error
After building an android app from the defold editor and launching it on a physical device, the app immediately crashes if its name contains numbers

**Steps to reproduce**
Steps to reproduce the behavior in the name of the project in game.project write numbers at the beginning. In our case it was "21 points".

**Expected behavior**
Build the apk app and run it on your smartphone

**Default version:**.
  - Version [1.4.1].

**Platforms:**.
 - Platforms: Android

**Screenshots (OPTION):**.
Screenshot of the error 
photo_2023-01-26_20-47-06

### PR DESCRIPTION
**Pull request by britzl**
An Android package name and Apple bundle identifier follows strict rules, but there are no checks in place to verify that the package name and bundle identifiers in `game.project` actually follow the rules. This fix adds a check when bundling for Android, iOS and macOS that these values follow their respective formats.

**Android**: Must consist of two or more segments separated by a dot. Each segment must start with a letter. Each segment must only consist of alphanumeric letters or the underscore character.

**iOS and macOS**: Must consist of two or more segments separated by a dot. Each segment must start with a letter. Each segment must only consist of alphanumeric letters, the underscore or hypen (-) character.

Fixes #7324

### CODE REVIEW

**Comment by britzl** [comment]
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java [hunk 1]
```diff
@@ -902,4 +902,23 @@ public static boolean isArchiveExcluded(Project project) {
...
+        return packageName.matches("^[a-zA-Z][a-zA-Z0-9_]*(\\.[a-zA-Z][a-zA-Z0-9_]*)+$");
+    }
+
+
+    // must have two or more segments
+    // each segment must start with a letter
+    // all characters must be alphanumeric, an underscore or a hyphen
+    public static boolean isValidAppleBundleIdentifier(String bundleIdentifier) {
+        return bundleIdentifier.matches("^[a-zA-Z][a-zA-Z0-9_-]*(\\.[a-zA-Z][a-zA-Z0-9_-]*)+$");
+    }
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:919]: I see these functions are isolated. So we probably will be able to reuse them from the editor in the future.

**Review by AGulev — APPROVED** [review]

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java (hunk 1 above)
Comment (britzl) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:919]: Yes, we should probably look into field validation for these in the editor. I'll take a quick look before merging this.

Comment (britzl) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:919]: We currently have no support for field validation in game.project. This fix will have to be for bob only for now.

**Pull request merged**